test/e2e/storage/vsphere_stress.go

```diff
@@ -0,0 +1,126 @@
+		// Volumes will be provisioned with each different types of Storage Class, in this case 50 volumes per SC.
+
+		scArrays := make([]*storageV1.StorageClass, 4)
+		// Create default vSphere Storage Class
+		By("Creating Storage Class : sc-default")
+		scDefaultSpec := getVSphereStorageClassSpec("sc-default", nil)
+		scDefault, err := client.StorageV1().StorageClasses().Create(scDefaultSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scDefault.Name, nil)
+		scArrays[0] = scDefault
+
+		// Create Storage Class with vSan Parameters
+		By("Creating Storage Class : sc-vsan")
+		var scVSanParameters map[string]string
+		scVSanParameters = make(map[string]string)
+		scVSanParameters[Policy_HostFailuresToTolerate] = "1"
+		scVSanSpec := getVSphereStorageClassSpec("sc-vsan", scVSanParameters)
+		scVSan, err := client.StorageV1().StorageClasses().Create(scVSanSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scVSan.Name, nil)
+		scArrays[1] = scVSan
+
+		// Create Storage Class with SPBM Policy
+		By("Creating Storage Class : sc-spbm")
+		var scSPBMPolicyParameters map[string]string
+		scSPBMPolicyParameters = make(map[string]string)
+		goldPolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scSPBMPolicyParameters[SpbmStoragePolicy] = goldPolicy
+		scSPBMPolicySpec := getVSphereStorageClassSpec("sc-spbm", scSPBMPolicyParameters)
+		scSPBMPolicy, err := client.StorageV1().StorageClasses().Create(scSPBMPolicySpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scSPBMPolicy.Name, nil)
+		scArrays[2] = scSPBMPolicy
+
+		// Create Storage Class with User Specified Datastore.
+		By("Creating Storage Class : sc-user-specified-ds")
+		var scWithDSParameters map[string]string
+		scWithDSParameters = make(map[string]string)
+		datastore := os.Getenv("VSPHERE_DATASTORE")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scWithDSParameters[Datastore] = datastore
+		scWithDatastoreSpec := getVSphereStorageClassSpec("sc-user-specified-ds", scWithDSParameters)
+		scWithDatastore, err := client.StorageV1().StorageClasses().Create(scWithDatastoreSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scWithDatastore.Name, nil)
+		scArrays[3] = scWithDatastore
+
+		instances, err := strconv.Atoi(os.Getenv("VCP_STRESS_INSTANCES"))
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP-STRESS-INSTANCES")
+
+		volumesPerInstance, err := strconv.Atoi(os.Getenv("VCP_STRESS_VOLUMES_PER_INSTANCE"))
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP_STRESS_VOLUMES_PER_INSTANCE")
+
+		var wg sync.WaitGroup
+		wg.Add(instances)
+		for instanceCount := 0; instanceCount < instances; instanceCount++ {
+			go CreateVolumesInParallel(client, namespace, scArrays, volumesPerInstance, &wg)
```

Comment:
Going further do we want to run full lifecycle of the PVC(s) in a thread like: create PVC, attach to pod, delete pod, delete PVC? 
If so then CreateVolumesInParallel is not good name for the function.

Reply:
Changed name to 
```
PerformVolumeLifeCycleInParallel
```